controllers/numaresourcesscheduler_controller.go

```diff
@@ -47,16 +74,110 @@ type NUMAResourcesSchedulerReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.9.2/pkg/reconcile
 func (r *NUMAResourcesSchedulerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	_ = context.Background()
+	klog.V(3).InfoS("Starting NUMAResourcesScheduler reconcile loop", "object", req.NamespacedName)
+	defer klog.V(3).InfoS("Finish NUMAResourcesScheduler reconcile loop", "object", req.NamespacedName)
+
+	instance := &nrsv1alpha1.NUMAResourcesScheduler{}
+	err := r.Get(context.TODO(), req.NamespacedName, instance)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
 
-	// your logic here
+	result, condition, err := r.reconcileResource(ctx, instance)
+	if condition != "" {
+		// TODO: use proper reason
+		reason, message := condition, messageFromError(err)
+		if err := r.updateStatus(ctx, r.Client, instance, condition, reason, message); err != nil {
+			klog.InfoS("Failed to update numaresourcesscheduler status", "Desired condition", condition, "error", err)
+		}
+	}
+	return result, err
 
-	return ctrl.Result{}, nil
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *NUMAResourcesSchedulerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&nodetopologyv1alpha1.NUMAResourcesScheduler{}).
+		For(&nrsv1alpha1.NUMAResourcesScheduler{}).
 		Complete(r)
 }
+
+func (r *NUMAResourcesSchedulerReconciler) reconcileResource(ctx context.Context, instance *nrsv1alpha1.NUMAResourcesScheduler) (reconcile.Result, string, error) {
+	klog.Info("SchedulerSync start")
+
+	deplomentInfo, err := r.syncNUMASchedulerResources(ctx, instance)
+	if err != nil {
+		return ctrl.Result{}, status.ConditionDegraded, errors.Wrapf(err, "FailedSchedulerSync")
+	}
+
+	instance.Status.Deployment = []nrsv1alpha1.NamespacedName{}
```

Comment:
we will have exactly one deployment, so let's simplify this code.
I'm not sure we need this feature, but since it's already there, let's keep it.
Let's NOT add more features to this API and controller - we need to support and test them, and this controller has a fixed and limited timespan. Hardly worth the extra effort.

Reply:
Additionally, did we extend the types definition to add this field to the `Status`?

Reply:
Yes we do https://github.com/Tal-or/numaresources-operator/blob/numares-sched-controller/api/numaresourcesoperator/v1alpha1/numaresourcesscheduler_types.go#L32

Reply:
please change the `Status.Deployment` to be only a single object - not a slice

Reply:
Is this change should be manually or I should use the operator-sdk framework for that?

Reply:
The way it works best for you. I usually edit the struct and then regenerate the artifacts out of it (`make generate` , `make manifests`, `make bundle`...)